### PR TITLE
Replace the deprecated CountableRange with Range

### PR DESCRIPTION
**Pull request by hartbit**

### CODE REVIEW

**Comment by hartbit** [comment]
@swift-ci please smoke test

**Comment by aciidb0mb3r** [comment]
We need to maintain compatibility with Swift 4.0 in Basic and Utility

**Comment by hartbit** [comment]
@aciidb0mb3r I added swift version checks

File: Sources/Basic/OrderedSet.swift [hunk 1]
```diff
@@ -13,7 +13,11 @@
 public struct OrderedSet<E: Hashable>: Equatable, Collection {
     public typealias Element = E
     public typealias Index = Int
+    #if swift(>=4.2)
+    public typealias Indices = Range<Int>
+    #else
     public typealias Indices = CountableRange<Int>
+    #endif
```
Comment (aciidb0mb3r) [inline Sources/Basic/OrderedSet.swift]: @jrose-apple is this check enough to not break compatibility with 4.0?

Comment (jrose-apple) [reply Sources/Basic/OrderedSet.swift]: Should be fine, but shouldn't that be checking for 4.1, not 4.2?

**Comment by hartbit** [comment]
@swift-ci please smoke test

**Pull request merged**